src/commands/flat.rs

```diff
@@ -0,0 +1,82 @@
+use std::io;
+use std::path::Path;
+
+use crate::config::option::DisplayOption;
+use crate::context::AppContext;
+use crate::error::JoshutoResult;
+use crate::fs::{JoshutoDirEntry, JoshutoDirList, JoshutoMetadata};
+
+use walkdir::WalkDir;
+
+fn _is_hidden(entry: &walkdir::DirEntry) -> bool {
+    entry
+        .file_name()
+        .to_str()
+        .map(|s| s.starts_with('.'))
+        .unwrap_or(false)
+}
+
+pub fn _walk_directory(
+    path: &Path,
+    options: &DisplayOption,
+    depth: usize,
+) -> io::Result<Vec<JoshutoDirEntry>> {
+    let results: Vec<JoshutoDirEntry> = WalkDir::new(path)
+        .max_depth(depth)
+        .into_iter()
+        .filter_entry(|e| {
+            if options.show_hidden() {
+                true
+            } else {
+                !_is_hidden(e)
+            }
+        })
+        .filter(|e| {
+            !e.as_ref()
+                .ok()
+                .unwrap()
```

Comment:
This crashes
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/commands/flat.rs:37:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
I think the cause for this is when you don't have enough permissions for specific folders. ie. `/root`.

Reply:
fixed